ImageViewer.php

```diff
@@ -285,14 +252,14 @@ function renderPreview($instrument, $record, $event_id, $repeat_instance, $proje
         }
 
         // We need to know the filetype to validate when the file has been previously uploaded...
-        /*$q = REDCap::getData('json',$record, array_keys($fields), $event_id);
-        *$results = json_decode($q, true);
-        *$result = $results[0];
-        */
+        $q = REDCap::getData('json',$record, array_keys($fields), $event_id);
+        $results = json_decode($q, true);
+        $result = $results[$repeat_instance - 1];
```

Comment:
That's interesting.. I haven't played around that much with repeat instances, but worry about a scenario where someone has three repeat instances and then deletes instance '2'.  Now, they go to do a preview when viewing instance 3.  In this case, isn't the index of the results array going to be off?  

I think one would have to actually iterate for a match to handle this scenario... @jwalsoe

Reply:
You are absolutely on the money with that worry. That is why I closed the pull request. The SQL lookup works fine though. I was testing more around repeat instances and found a bug in RC yesterday, which I rapported in, seems to be a case of ticket #35814 which still has a problem when first instance is deleted. Also has consequences for file upload fields in other instances,
 where file reference gets out of whack (deleting instances does not completely remove it from dash).